[Banner] Added BannerCfg class to hostcfgd to handle Banner messages

SviatoslavBoichuk opened the pull request:
#### Why I did it
Added Banner feature related hostcfgd service according to HLD: https://github.com/sonic-net/SONiC/pull/1361

This PR depends on the following PR: https://github.com/sonic-net/sonic-swss-common/pull/826

#### How I did it
Added BannerCfg service to hostcfgd. It handles Config DB changes and restart "banner-config" systemd service which will update banners.

#### How to verify it
Manual test

liat-grozovik commented:
@SviatoslavBoichuk  please handle conflict. Please also add description of the code changes and motivation

SviatoslavBoichuk commented:
> @SviatoslavBoichuk please handle conflict. Please also add description of the code changes and motivation

Fixed conflicts, updated PR description

qiluo-msft commented on scripts/hostcfgd:
```diff
@@ -1500,6 +1500,85 @@ class FipsCfg(object):
+            key: DB table's key that was triggered change.
+            data: Read table data.
+        """
+        # Handling state, login/logout and MOTD messages. Data should be a dict
```
>state [](http://example.com/codeflow?start=19&length=5)

Where is the code to handle `state`? If the feature is disabled, we expected that no extra daemon started, minimal CPU wasted in this script.

SviatoslavBoichuk replied:
The "state" is used by "banner-config.service". Hostcfgd responsible only to track DB changes for BANNER table and restart banner service once user modified banner messages or feature state. The banner service (please see PR: https://github.com/sonic-net/sonic-buildimage/pull/16957/) will apply new Banners to Linux only if feature state == enabled.

qiluo-msft replied:
How to disable the feature? In normal deployment, if people do not want use this feature, would like to disable it completely, minimize teh CPU wasted relating to this feature.

SviatoslavBoichuk replied:
To disable the feature - you can use CLI command: config banner state disabled.

qiluo-msft replied:
If the feature is disabled, we expected that no extra daemon started, minimal CPU wasted in this script. In this case, is the service "banner-config" still running?

fastiuk replied:
@qiluo-msft is that thread resolved?

liat-grozovik commented:
/azp run sonic-net.sonic-host-services

fastiuk commented:
The test is failing because https://github.com/sonic-net/sonic-swss-common/pull/826 should be merged first

fastiuk commented:
Still failing because of https://github.com/sonic-net/sonic-swss-common/pull/826

fastiuk commented:
@qiluo-msft , added a new unit test. Didn't touch a code. Please re-review/approve

fastiuk commented:
@qiluo-msft , gentle reminder :)

fastiuk commented:
@qiluo-msft , gentle reminder :)
Please approve if you have time

jhli-cisco commented:
Hi @SviatoslavBoichuk we are interested in the banner feature. can it be merged anytime soon?

qiluo-msft commented:
Please resolve the conflict

fastiuk commented:
@qiluo-msft rebased. Please merge

fastiuk commented:
@qiluo-msft could you please merge it? It was approved by you, I rebased and it passed the CI here

fastiuk commented:
@qiluo-msft gentle reminder to merge
